pkg/server/server.go

```diff
@@ -0,0 +1,254 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/gofiber/fiber/v2"
+	"github.com/gofiber/fiber/v2/middleware/basicauth"
+	"github.com/gofiber/fiber/v2/middleware/compress"
+	"github.com/gofiber/fiber/v2/middleware/cors"
+	"github.com/gofiber/fiber/v2/middleware/logger"
+	"github.com/gofiber/fiber/v2/middleware/recover"
+	"github.com/rotisserie/eris"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/G-Research/fasttrackml/pkg/api/admin/service/namespace"
+	aimAPI "github.com/G-Research/fasttrackml/pkg/api/aim"
+	mlflowAPI "github.com/G-Research/fasttrackml/pkg/api/mlflow"
+	mlflowConfig "github.com/G-Research/fasttrackml/pkg/api/mlflow/config"
+	mlflowController "github.com/G-Research/fasttrackml/pkg/api/mlflow/controller"
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/dao"
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/dao/repositories"
+	mlflowRepositories "github.com/G-Research/fasttrackml/pkg/api/mlflow/dao/repositories"
+	mlflowService "github.com/G-Research/fasttrackml/pkg/api/mlflow/service"
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/service/artifact"
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/service/artifact/storage"
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/service/experiment"
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/service/metric"
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/service/model"
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/service/run"
+	namespaceMiddleware "github.com/G-Research/fasttrackml/pkg/common/middleware/namespace"
+	"github.com/G-Research/fasttrackml/pkg/database"
+	adminUI "github.com/G-Research/fasttrackml/pkg/ui/admin"
+	adminUIController "github.com/G-Research/fasttrackml/pkg/ui/admin/controller"
+	aimUI "github.com/G-Research/fasttrackml/pkg/ui/aim"
+	"github.com/G-Research/fasttrackml/pkg/ui/chooser"
+	chooserController "github.com/G-Research/fasttrackml/pkg/ui/chooser/controller"
+	mlflowUI "github.com/G-Research/fasttrackml/pkg/ui/mlflow"
+	"github.com/G-Research/fasttrackml/pkg/version"
+)
+
+type Server interface {
+	Listen(address string) error
+	ShutdownWithTimeout(timeout time.Duration) error
+	Test(req *http.Request, msTimeout ...int) (*http.Response, error)
+}
+
+type server struct {
+	*fiber.App
+}
+
+func NewServer(ctx context.Context, config *mlflowConfig.ServiceConfig) (Server, error) {
+	// init database connection.
+	db, err := initDB(config)
+	if err != nil {
+		return nil, err
+	}
+
+	var namespaceRepository repositories.NamespaceRepositoryProvider
+	var artifactStorageFactory storage.ArtifactStorageFactoryProvider
+	if err := func() error {
```

Comment:
this `if` guy looks a bit strange. Im understand why but in general can we maybe do a bit differenttly?

Reply:
maybe we can create `db` in  `serverCmd` and pass it into `NewServer` function. Then we can analyse `err` from `NewServer` and just close `db` in that case. So in that case everything could look a bit more understandable.

Reply:
good point - I have tried yet another approach of create the namespace repository and its listener in a separate function, which hopefully makes the whole thing simpler to grasp.